### ISSUE TITLE
Process.MainWindowTitle and Process.Responding are not correctly refreshed

### ISSUE THREAD

**Issue by JeroenOortwijn** [dotnet/runtime#36768]
As a followup on issue #32690, both the properties `Process.MainWindowTitle` and `Process.Responding` are also not properly refreshed.

I have a test where I check if a program has loaded a file by checking if the filename is present in the title of the program:
```
private bool IsDoneLoading()
{
    _roboguide.Refresh();
    return _roboguide.MainWindowTitle.Contains(_workCellName);
}
```
While looping over this, after a while the function returns `true` in .NET Framework, but stays `false` in .Net 5 preview 4.

The fix for this would be similar to the one in PR #32695:
In [RefreshCore(),](https://github.com/dotnet/runtime/blob/9d9008ee5ec979543c997616f33de5283b73bad5/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L116) add `_mainWindowTitle = null;`.

While looking through the code, I noticed that `Process.Responding` might also have the same issue.
The fix for this would be:
In [RefreshCore(),](https://github.com/dotnet/runtime/blob/9d9008ee5ec979543c997616f33de5283b73bad5/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L116) add `_haveResponding = false;`.

**Comment by JeroenOortwijn** [comment]
I've made a test showing the `Process.MainWindowTitle` bug at: https://github.com/JeroenOortwijn/MainWindowTitle-test

**Comment by danmosemsft** [comment]
@JeroenOortwijn thanks for pointing this out. You are right. Do you want to offer a PR, similar to #32695?

**Comment by eiriktsarpalis** [comment]
Fixed by #36768
